resources/js/Components/LoadingOverlay.vue

```diff
@@ -14,8 +14,8 @@ import { ref } from 'vue';
 import { CdxProgressBar } from "@wikimedia/codex";
 
 const props = withDefaults(defineProps<{
-	delay: number
-	visible: boolean
+	delay?: number
+	visible?: boolean
```

Comment:
Isn't making these optional kind of risky? If we somehow don't pass `visible` the `show()` method breaks. Is there a way around this that leaves them required?

Reply:
They have default values, therefore they won't be `null` or `undefined`

Reply:
> They have default values, therefore they won't be `null` or `undefined`

I also had this issue, i was getting a TS warning or error because of this at some point but i dont see it anymore and they do have defaults, but my question is... is this part of the scope of this ticket? :grimacing:

Reply:
Valid question :D Vue documentation says that you should remove all the warnings before removing the compat build, I took it as a sign :D

Reply:
> Valid question :D Vue documentation says that you should remove all the warnings before removing the compat build, I took it as a sign :D

ah ok, so you got the warning.. i also saw it. yes, we shouldn't merge code with warnings :)

Reply:
> Isn't making these optional kind of risky? If we somehow don't pass `visible` the `show()` method breaks. Is there a way around this that leaves them required?

@chukarave the only instance of the loading overlay we are using in the app is not passing the props and this is the reason we are getting the warnings `<loading-overlay ref="overlayRef" />` https://github.com/wmde/wikidata-mismatch-finder/blob/development/resources/js/Pages/Results.vue#L3 so yeah, the right call is to make them optional, since they are not being passed anyway

Reply:
Right those were exactly the warnings we were getting. Nice to have a clean console :raised_hands: well done @hasanakg